cmd/root.go

```diff
@@ -163,6 +165,107 @@ func parseToolsFile(ctx context.Context, raw []byte) (ToolsFile, error) {
 	return toolsFile, nil
 }
 
+func parse(ctx context.Context, buf []byte, logger log.Logger) {
+	logger.DebugContext(ctx, "Attempting to parse updated tools file.")
+	// time.Sleep(5 * time.Second)
+	// logger.DebugContext(ctx, "Parse finished.")
+	// TODO: add logic for parsing
+}
+
+// watchFile checks for changes in the provided yaml tools file.
+func watchFile(toolsFileName string, ctx context.Context, logger log.Logger) {
+	w, err := fsnotify.NewWatcher()
+	if err != nil {
+		logger.WarnContext(ctx, "error setting up new watcher %s", err)
+		return
+	}
+
+	defer w.Close()
+
+	err = w.Add(filepath.Dir(toolsFileName))
+	if err != nil {
+		logger.WarnContext(ctx, "error adding the tools file to watcher %s", err)
+	}
+
+	logger.DebugContext(ctx, fmt.Sprintf("Now watching tools file %s", toolsFileName))
+
+	// when the user saves a file, we wait for 100ms of no new events (due to how many editors perform writes)
+	debounceDelay := 100 * time.Millisecond
+
+	var timerC <-chan time.Time
+	var debounceTimer *time.Timer
+
+	defer func() {
+		if debounceTimer != nil {
+			debounceTimer.Stop()
+		}
+	}()
+
+	for {
+		select {
+		case <-ctx.Done():
+			// potentially unnecessary due to defer above?
+			if debounceTimer != nil {
+				debounceTimer.Stop()
+			}
+			logger.DebugContext(ctx, "file watcher context cancelled")
+			return
+
+		case err, ok := <-w.Errors:
+			if !ok {
+				logger.WarnContext(ctx, "file watcher alredy closed")
+				return
+			}
+			if err != nil {
+				logger.WarnContext(ctx, "file watcher error %s", err)
+				return
+			}
+
+		case e, ok := <-w.Events:
+			if !ok {
+				logger.WarnContext(ctx, "file watcher already closed")
+				return
+			}
+
+			if filepath.Clean(e.Name) == filepath.Clean(toolsFileName) && e.Op == fsnotify.Write {
```

Comment:
nit: should save the output from `filepath.Clean(toolsFileName)` somewhere so it doesn't need to run every time